### ISSUE TITLE
FieldResolver in abstract resolver throws an error

### ISSUE THREAD

**Issue by limenutt** [19majkel94/type-graphql#102]
Resolver inheritance feature is great, helps to reduce the amount of boilerplate code!

Currently, if you add a `@FieldResolver` to the abstract resolver it throws an error 
_No provided object type in '`@Resolver`' decorator for class AbstractResolverClassName!'_
The easiest way to reproduce this would be to add a field resolver to the abstract resolver from the examples/resolvers-inheritance
```
@FieldResolver()
    protected async resourceName() {
      return "dummy string";
    }
```
I wonder if there is a way to add a field resolver there or the error is by design?

**Comment by 19majkel94** [comment]
You need to provide `of => objectTypeClass` parameter, like with a normal resolver class - check the decorator signature.

**Comment by limenutt** [comment]
Oh, I see. For some reason, I thought that since the Resolver was marked with `{ isAbstract: true }`, it did not require a type, since one would be provided in the concrete implementation of this abstract resolver. Since it worked without type until I've added the field resolver to it, I decided it had something to do with field resolver. 
Thank you, after I've added the type to the abstract resolver, it works fine now.

I'd suggest updating the resolver inheritance example by supplying resolver type there as well.

**Comment by 19majkel94** [comment]
> Since it worked without type until I've added the field resolver to it, I decided it had something to do with field resolver.

It always worked like that - your queries and mutations don't need to be attached to specific `ObjectType` but `FieldResolver` has to. Just like it's stated in docs:
https://19majkel94.github.io/type-graphql/docs/resolvers.html#field-resolvers

**Comment by 19majkel94** [comment]
Fixed via 6655ac1 :tada
